tool/tsh/tsh.go

```diff
@@ -1305,7 +1305,9 @@ func onLogin(cf *CLIConf) error {
 		}
 	}
 
+	usernameProvided := true
 	if cf.Username == "" {
+		usernameProvided = false
```

Comment:
Is this assumption correct? What if `cf.Username == ""` and `tc.Username == ""` too?

Reply:
See above comment about `ExplicitUsername`.